R/setup.R

```diff
@@ -0,0 +1,37 @@
+#' setup
+#'
+#' Prepare rOpenscmRunner to run climate models.
+#'
+#' Some simple climate models need certain information to run. This function provides a unified
+#' interface to supply this information.
+#'
+#' @param magiccExecutable7 The file path of a MAGICC version 7 executable file (with corresponding config file
+#' structure around it). You have to supply this if you intend to run MAGICC version 7. Instead of using this
+#' function to supply it, you can also set the environment variable MAGICC_EXECUTABLE_7 before starting your R session.
+#' @param magiccWorkerNumber The number of processes which should be started when running MAGICC. By default, as many
+#' processes are started as there are processors in your system. Instead of using this function to supply the setting,
+#' you can also set the environment variable MAGICC_WORKER_NUMBER before starting your R session.
+#' @param magiccWorkerRootDir The path to a folder where the temporary directories to run MAGICC will be created.
+#' By default, a standard location is chosen depending on your operating system. Instead of using this function to
+#' supply the setting, you can also set the environment variable MAGICC_WORKER_ROOT_DIR before starting your R session.
+#' If you do supply a non-standard directory, you have to create and clean it up yourself. If you use the standard
+#' settings, everything happens automatically.
+#' @author Mika Pflüger
+#' @importFrom reticulate import
+#' @examples
+#' setup(magiccExecutable7 = "/path/to/magicc/bin/magicc", magiccWorkerNumber = 4)
+#' @export
+setup <- function(magiccExecutable7 = NULL, magiccWorkerNumber = NULL, magiccWorkerRootDir = NULL) {
+  if (!is.null(magiccExecutable7) || !is.null(magiccWorkerNumber) || !is.null(magiccWorkerRootDir)) {
+    os <- import("os")
+    if (!is.null(magiccExecutable7)) {
+      os$environ$"__setitem__"("MAGICC_EXECUTABLE_7", magiccExecutable7)
```

Comment:
```suggestion
      os$environ[["MAGICC_EXECUTABLE_7"]] <- magiccExecutable7
```
Does this not work? Or maybe with single instead of double `[]`?

Reply:
Is it important to do this via reticulate? In R it would be this:
```r
Sys.setenv(MAGICC_EXECUTABLE_7 = magiccExecutable7)
```

Reply:
Yes, we need it in the python side of things, and the python side of things only gets the R environment variables when reticulate is loaded the first time, therefore updating the R side of things is unreliable.

Reply:
> Does this not work? Or maybe with single instead of double []?

I originally tried with single `[` and got an error I didn't understand, which is why I dodged using `__setitem__`. Double quotes work, indeed.

Reply:
Nope, double brackets only work for non-existing values. Overwriting existing values doesn't work with double-bracket-assignment. Don't ask me why, I just reverted to `__setitem__`.